src/key.rs

```diff
@@ -168,9 +168,13 @@ impl str::FromStr for PublicKey {
     fn from_str(s: &str) -> Result<PublicKey, Error> {
         let mut res = [0u8; constants::UNCOMPRESSED_PUBLIC_KEY_SIZE];
         match from_hex(s, &mut res) {
-            Ok(constants::PUBLIC_KEY_SIZE) =>
-                PublicKey::from_slice(&res[0..constants::PUBLIC_KEY_SIZE]),
-            Ok(constants::UNCOMPRESSED_PUBLIC_KEY_SIZE) => PublicKey::from_slice(&res),
+            Ok(constants::PUBLIC_KEY_SIZE) => {
+                let bytes: [u8; constants::PUBLIC_KEY_SIZE] =
+                    res[0..constants::PUBLIC_KEY_SIZE].try_into().unwrap();
```

Comment:
Strictly speaking, it'd be better to create a reference here (simply annotating the type with `&` should work because the conversion is defined) to avoid useless copying but there's a good chance compiler will optimize it out anyway.

Reply:
Are you saying you can convert a `&[u8]` to a `&[u8; N]` via `try_into` and the compiler will just cast the pointers rather than copying?

Reply:
Exactly.

Reply:
Wow, that's a huge win and I feel like a clippy lint should exist for it. (Though it'd be a bit tough in general to tell when the array is only used by reference.)